tests/garage/tf/models/test_model.py

```diff
@@ -0,0 +1,178 @@
+
+        model2 = SimpleModel(output_dim=2)
+        with self.assertRaises(ValueError):
+            model2.build(another_input_var)
+
+    def test_model_with_different_name(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        another_input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = SimpleModel(output_dim=2)
+        outputs_1 = model.build(input_var)
+        outputs_2 = model.build(another_input_var, name='network_2')
+        data = np.ones((3, 5))
+        results_1, results_2 = self.sess.run([outputs_1, outputs_2],
+                                             feed_dict={
+                                                 input_var: data,
+                                                 another_input_var: data
+                                             })  # noqa: E126
+        assert np.array_equal(results_1, results_2)
+
+    def test_model_with_different_name_in_different_order(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        another_input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = SimpleModel(output_dim=2)
+        outputs_1 = model.build(input_var, name='network_1')
+        outputs_2 = model.build(another_input_var)
+        data = np.ones((3, 5))
+        results_1, results_2 = self.sess.run([outputs_1, outputs_2],
+                                             feed_dict={
+                                                 input_var: data,
+                                                 another_input_var: data
+                                             })  # noqa: E126
+        assert np.array_equal(results_1, results_2)
+
+    def test_model_in_model(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = ComplicatedModel(output_dim=2)
+        outputs = model.build(input_var)
+        data = np.ones((3, 5))
+        out, model_out = self.sess.run(
+            [outputs, model.networks['default'].outputs],
+            feed_dict={model.networks['default'].input: data})
+        assert np.array_equal(out, model_out)
+
+    def test_model_pickle(self):
+        data = np.ones((3, 5))
+        model = SimpleModel(output_dim=2)
+
+        with tf.Session(graph=tf.Graph()) as sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model.build(input_var)
+
+            results = sess.run(
+                model.networks['default'].outputs,
+                feed_dict={model.networks['default'].input: data})
+            model_data = pickle.dumps(model)
+
+        with tf.Session(graph=tf.Graph()) as sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model_pickled = pickle.loads(model_data)
+            outputs = model_pickled.build(input_var)
+
+            results2 = sess.run(outputs, feed_dict={input_var: data})
+
+        assert np.array_equal(results, results2)
+
+    def test_model_in_model_pickle_same_parameters(self):
```

Comment:
this looks like `test_complicated_model_pickle` to me. where are the `same_parameters`?